File: ios/Classes/model/FlutterOSSClientConfiguration.swift
//
//  OSSClientConfiguration.swift
//  flutter_oss_sdk
//
//  Created by kangkang on 2023/2/22.
//

import Foundation

struct FlutterOSSClientConfiguration {
    /// STS应用服务器地址
    var ossStsUrl: String

    /// 阿里云OSS服务
    var bucketEndPoint: String

    /// 设置是否开启DNS配置
    var httpDnsEnable: Bool

    /// 设置自定义user-agent
    var userAgentMark: String?

    /// 设置网络参数
    /// 连接超时，默认15秒
    var connectionTimeout: Int

    /// socket超时，默认15秒
    var socketTimeout: Int

    /// 最大并发请求书，默认5个
    var maxConcurrentRequest: Int

    /// 失败后最大重试次数，默认2次
    var maxErrorRetry: Int

    /// 是否开启开启日志
    var enableLog: Bool
    
    /**
     init(arguments:[String: Any]){
         bucketName = arguments["bucketName"] as? String ?? "";
         objectKey = arguments["objectKey"] as? String ?? "";
         uploadFilePath = arguments["uploadFilePath"] as? String ?? "";
         customDomain = arguments["customDomain"] as? String;
         callbackUrl = arguments["callbackUrl"] as? String;
         callbackHost = arguments["callbackHost"] as? String;
         callbackBodyType = arguments["callbackBodyType"] as? String;
         callbackBody = arguments["callbackBody"] as? String;
         callbackVars = arguments["callbackVars"] as? [String: String?];
     }
     */
    
    init(arguments:[String: Any]){
        ossStsUrl = arguments["ossStsUrl"] as? String ?? "";
        bucketEndPoint = arguments["bucketEndPoint"] as? String ?? "";
        httpDnsEnable = arguments["httpDnsEnable"] as? Bool ?? false;
        userAgentMark = arguments["userAgentMark"] as? String;
        connectionTimeout = arguments["connectionTimeout"] as? Int ?? 15 * 1000;
        socketTimeout = arguments["socketTimeout"] as? Int ?? 15 * 1000;
        maxConcurrentRequest = arguments["maxConcurrentRequest"] as? Int ?? 5;
        maxErrorRetry = arguments["maxErrorRetry"] as? Int ?? 3;
        enableLog = arguments["enableLog"] as? Bool ?? true;
    }
}

File: android/src/main/java/com/example/flutter_oss_sdk/model/FlutterOSSClientConfiguration.java
package com.example.flutter_oss_sdk.model;

public class FlutterOSSClientConfiguration {
    ///STS应用服务器地址
    private String ossStsUrl;

    ///阿里云OSS服务
    private String bucketEndPoint;

    ///设置是否开启DNS配置
    private Boolean httpDnsEnable = true;

    ///设置自定义user-agent
    private String userAgentMark;

    ///设置网络参数
    ///连接超时，默认15秒
    private Integer connectionTimeout;

    ///socket超时，默认15秒
    private Integer socketTimeout;

    ///最大并发请求书，默认5个
    private Integer maxConcurrentRequest;

    ///失败后最大重试次数，默认2次
    private Integer maxErrorRetry;


    ///是否开启开启日志
    private Boolean enableLog;

    public void setOssStsUrl(String ossStsUrl) {
        this.ossStsUrl = ossStsUrl;
    }

    public void setBucketEndPoint(String bucketEndPoint) {
        this.bucketEndPoint = bucketEndPoint;
    }

    public void setHttpDnsEnable(Boolean httpDnsEnable) {
        this.httpDnsEnable = httpDnsEnable;
    }

    public void setUserAgentMark(String userAgentMark) {
        this.userAgentMark = userAgentMark;
    }

    public void setConnectionTimeout(Integer connectionTimeout) {
        this.connectionTimeout = connectionTimeout;
    }

    public void setSocketTimeout(Integer socketTimeout) {
        this.socketTimeout = socketTimeout;
    }

    public void setMaxConcurrentRequest(Integer maxConcurrentRequest) {
        this.maxConcurrentRequest = maxConcurrentRequest;
    }

    public void setMaxErrorRetry(Integer maxErrorRetry) {
        this.maxErrorRetry = maxErrorRetry;
    }

    public void setEnableLog(Boolean enableLog) {
        this.enableLog = enableLog;
    }

    public Boolean getEnableLog() {
        return enableLog;
    }

    public String getOssStsUrl() {
        return ossStsUrl;
    }

    public String getBucketEndPoint() {
        return bucketEndPoint;
    }

    public Boolean getHttpDnsEnable() {
        return httpDnsEnable;
    }

    public String getUserAgentMark() {
        return userAgentMark;
    }

    public Integer getConnectionTimeout() {
        return connectionTimeout;
    }

    public Integer getSocketTimeout() {
        return socketTimeout;
    }

    public Integer getMaxConcurrentRequest() {
        return maxConcurrentRequest;
    }

    public Integer getMaxErrorRetry() {
        return maxErrorRetry;
    }
}
